composer/utils/checkpoint.py

```diff
@@ -276,9 +278,14 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                 # PyTorch will capture any exception of this function,
                 # and dist.all_gather_objects(exception) before raising it.
                 # If that all_gather_objects fails, the exception is never visible to user.
-                # We immediately print the exception to avoid that situation.
-                log.error(f'Exception {type(e)} raised during downloading: {str(e)}')
-                raise e
+                # We immediately kill the process and print the exception 
+                import psutil
+                log.warning(f'Exception {type(e)} raised during downloading: {str(e)}, terminating the process')
+                pid = os.getpid()
+                proc = psutil.Process(pid)
+                for child_process in proc.children(recursive=True):
+                    os.kill(child_process.pid, signal.SIGKILL)
+                os.kill(pid, signal.SIGKILL)
```

Comment:
This kills the current process, but it wont kill other processes which will still NCCL deadlock, right?

This shoudl send a SIGTERM to the launcher process?

Reply:
yeah, Irene has a PR to kill all processes https://github.com/mosaicml/composer/pull/3345